### DISCUSSION TITLE
Enable up-follow for all links inside element

### DISCUSSION THREAD

**Discussion by stefanfisk** [unpoly/unpoly#384] [Q&A] [answered]
Is it possible to enable `up-follow` for all links inside an element without using `up.link.config.followSelectors`? I tried `<div up-follow up-target="#my-container">`, but unsurprisingly it did nothing.

**Comment by adam12** [comment, marked as answer]
You could do it with an [`up.macro`](https://unpoly.com/up.macro), but you'll have to copy the attributes you want from the parent to each link.

**Reply by stefanfisk** [reply]
I'd completely forgotten about macros! Implementing one for this feature was just a few lines of code.

**Comment by triskweline** [comment]
Expanding on @adam12's idea we could probably build an `[app-subtree-target]` attribute like this:

```html
<div up-follow app-subtree-target="#my-container">
```

```js
up.compiler('[app-subtree-target]', function(element) {
  let target = element.getAttribute('app-subtree-target')
  up.on(element, 'up:click', 'a', function(event, link) {
    event.preventDefault()
    up.follow(link)
  })
})
```

Untested.
